File: examples/canonical-iac-pulumi/dotnet/azure/components/function-component/FunctionComponent.cs
using Pulumi;
using Pulumi.AzureNative.Web;

public sealed class FunctionComponent : ComponentResource
{
    public FunctionComponent(string name, string environment, string namePrefix, Input<string> identityId, ComponentResourceOptions? opts = null)
        : base("accb:azure:FunctionComponent", name, opts)
    {
        var stackName = Deployment.Instance.StackName;
        _ = new WebApp($"{namePrefix}-{environment}-{stackName}-func", new WebAppArgs
        {
            Kind = "functionapp,linux"
        }, new CustomResourceOptions { Parent = this });
        RegisterOutputs();
    }
}

File: examples/canonical-iac-pulumi/python/aws/components/function_component.py
import pulumi


class FunctionComponent(pulumi.ComponentResource):
    def __init__(self, name: str, environment: str, name_prefix: str, identity_ref, opts=None):
        super().__init__("accb:aws:FunctionComponent", name, None, opts)
        stack = pulumi.get_stack()
        self.function_name = pulumi.Output.from_input(f"{name_prefix}-{environment}-{stack}-handler")
        self.register_outputs({"function_name": self.function_name})
